### PR TITLE
Add labels and associated features

### PR DESCRIPTION
**Pull request by JRegimbal**
The following features are added in this pull request:

* implementing resource labels with unique names
* allowing resource label names to be updated through the API
* ability to filter resources by associated labels (AND not OR for multiple labels)
* addition of "Labeler" job that adds a label to specified resources generated in a workflow run
* generation of a "snapshot" with an overview of the workflow run in its description

### CODE REVIEW

File: rodan/jobs/base.py [hunk 1]
```diff
@@ -924,6 +927,41 @@ def run(self, runjob_id):
                     ]
                 )
 
+                # Update workflow run description with job info
+                wall_time = time.time() - start_time
+                snapshot_info = '\n\n{0}:\n    name: "{1}"\n    wall_time: "{2}"\n'.format(
```
Comment (deepio) [inline rodan/jobs/base.py:932]: Put it all in double quotes. Either escape the double quotes or put the whole thing in a triple quote so you don't need to escape anything.

File: rodan/jobs/labeler.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            else:
+                raise RuntimeError(
+                    (
+                        "Cannot find any resource or resource list on Input" " {0}"
+                    ).format(input.uuid)
+                )
+        return inputs
+
+    def run_my_task(self, inputs, settings, outputs):
+        label_text = settings['Label'] if settings['Label'] != '' \
```
Comment (deepio) [inline rodan/jobs/labeler.py:93]: double quotes

File: rodan/jobs/labeler.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    enabled = True
+    category = 'Utility'
+    interactive = False
+
+    input_port_types = [
+        {
+            'name': 'Resource',
+            'minimum': 1,
+            'maximum': 100,
+            'resource_types': lambda mime: re.match(r"^[-\w]+/[-\w+]+$", mime)
```
Comment (deepio) [inline rodan/jobs/labeler.py:32]: Double quotes

File: rodan/views/resourcelabel.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+from rodan.models.resourcelabel import ResourceLabel
+from rodan.permissions import CustomObjectPermissions
+from rodan.serializers.resourcelabel import ResourceLabelSerializer
+from rodan.paginators.pagination import CustomPaginationWithDisablePaginationOption
+
+
+class ResourceLabelList(generics.ListAPIView):
+    """
+    """
+    permission_classes = (permissions.IsAuthenticated, CustomObjectPermissions)
```
Comment (deepio) [inline rodan/views/resourcelabel.py:16]: Good.

File: rodan/models/resourcelabel.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    #     related_name="label",
+    #     null=True,
+    #     blank=True,
+    #     on_delete=models.SET_NULL,
+    #     db_index=True,
+    # )
+
+    class Meta:
+        app_label = "rodan"
+        # permissions = (("view_resource", "View Resource"),)
```
[the remark this answers is not in the document]
Comment (JRegimbal) [reply rodan/models/resourcelabel.py:30]: Should this be something to add in now or later?

**Pull request merged**